### PR TITLE
Fix a panic in connect-inject when the provided upstreams list is malformed

### PR DESCRIPTION
**Pull request by dhiaayachi**
### Changes proposed in this PR ###  
- Fix a panic when parsing a malformed upstream, now a malformed upstream will be ignored
- Support multiple separators between upstreams (<space>, <comma>, <return>).

### How I've tested this PR ###
Added unit tests to test multiple scenarios including malformed upstream lists

### How I expect reviewers to test this PR ###
N/A

### Checklist ###
- [x] Tests added
- [x] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Review by wilkermichael — APPROVED** [review]
Thanks for fixing this :D! I think you're just missing a space separated case in your tests

**Comment by dhiaayachi** [comment]
> Thanks for fixing this :D! I think you're just missing a space separated case in your tests

[This](https://github.com/hashicorp/consul-k8s/blob/4c471dd4addd60f171c46b717d26648eff01f2c1/control-plane/connect-inject/webhook/container_env_test.go#L70-L70) test case should cover it

**Review by wilkermichael — APPROVED** [review]
Ah, fantastic!

File: .changelog/3956.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+```release-note:bug
+control-plane: fix a panic when an upstream annotation is malformed.
+```
+
+```release-note:enhancement
+control-plane: support <space>, <comma> and <\n> as upstream separators.
```
Comment (komapa) [inline .changelog/3956.txt:6]: ❤️

**Pull request merged**